flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -537,7 +543,6 @@ public YarnClusterClient createCluster(
 		Preconditions.checkNotNull(userJarFiles, "User jar files should not be null.");
 
 		AbstractYarnClusterDescriptor yarnClusterDescriptor = createDescriptor(applicationName, cmdLine);
-		yarnClusterDescriptor.setFlinkConfiguration(config);
```

Comment:
You cannot simply ignore the `configuration` which has been given to `createCluster`. What if it contains configuration values which were added programmatically?